### ISSUE TITLE
Define JSON transforms for union

### ISSUE THREAD

**Issue by abuiles** [abuiles/json-xdr#5]
For the following definition

```
union Memo switch (MemoType type)
{
case MEMO_NONE:
    void;
case MEMO_TEXT:
    string text<28>;
case MEMO_ID:
    uint64 id;
case MEMO_HASH:
    Hash hash; // the hash of what to pull from the content server
case MEMO_RETURN:
    Hash retHash; // the hash of the tx you are rejecting
};
```

https://github.com/stellar/js-xdr/issues/25 suggest a transform like the following:
 
```
memo = {
 type: MEMO_TEXT,
 text: "Hello world!"
}
```

In theory this should work, but it could become ambiguous easily, for example, what if the arm has struct which has `type` as a member?

I'm leaning towards a representation where we make the `discriminant` and its `arm` explicit. I'm using the XDR language but we could abstract that away and just call them `type` and `value`.

```
// discriminant MEMO_TEXT
memo = {
  discriminant: "MEMO_TEXT", // could also be the value int equivalent in the enum (for example 1)
  arm: {
    text: "Hello world!"
  }
}

// discriminant MEMO_NONE

memoNone = {
  discriminant: "MEMO_NONE"
  arm: undefined // arm is optional  since its value is void
}
```

**Comment by abuiles** [comment]
We could also use the following, which takes away XDR specific language.

```
type: "MEMO_TEXT",
value: {
  text: "Hello world!"
}
```

**Comment by morleyzhi** [comment]
Personally, I value understandable, intuitive names highly, so I'd cast a vote for "type" / "value"!

**Comment by tomquisel** [comment]
I like "type" / "value", too. It's pretty good for dot notation:

```js
if (txn.memo.type === "MEMO_TEXT") {
    txn.memo.value.text.startsWith(...)
}
```

The approach is slightly more awkward than the approach laid out in https://github.com/stellar/js-xdr/issues/25 but you're right that there's ambiguity if the RPC struct contains a `type` variable. Looking at the XDR [spec](https://tools.ietf.org/html/rfc4506#page-17):
> An
   identifier is a letter followed by an optional sequence of letters,
   digits, or underbar ('_').
So you could do this as an alternative and be sure that there will be no name collision:
```js
memo = {
 _type: "MEMO_TEXT",
 text: "Hello world!"
}
```

leading to:
```js
if (txn.memo._type === "MEMO_TEXT") {
    txn.memo.text.startsWith(...)
}
```

I think it's nice to cut down the extra level of nesting, they add up as the RPC structs can be deeply nested. Not gonna die on that cross, though 😄 . Having `_` in JS identifiers feels bad too.

**Comment by abuiles** [comment]
Implemented in #16
